### PR TITLE
Add kiosk mode like grafana so that other system can embed page

### PR DESCRIPTION
**Pull request by jeremyxu2010**
** Describe the change **

Add kiosk mode like grafana so that other system can embed page

** Issue reference **

* [#651](https://github.com/kiali/kiali/issues/651)

** Backwards compatible? **

[ X ] Is your pull-request introducing changes in behaviour?

** Screenshot **

Add a screenshot of the UI after your changes.
image

### CODE REVIEW

**Comment by pilhuhn** [comment]
@jeremyxu2010 Thank you for your contribution. Please help me to better understand it: Does the Kiosk mode restrict the display to the overview page only? Or if you can navigate from overview to the graph, how would the user get back to the overview?

**Comment by jeremyxu2010** [comment]
> @jeremyxu2010 Thank you for your contribution. Please help me to better understand it: Does the Kiosk mode restrict the display to the overview page only? Or if you can navigate from overview to the graph, how would the user get back to the overview?

Yes. Grafana's kiosk mode restrict the display to the target page. Grafana's content area does not contains any page link to other page, which is exactly what I need.

**Comment by jotak** [comment]
@pilhuhn I guess the navigation is going to be handled by the application where it's embedded

File: src/components/Nav/NavUtils.tsx [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import { GraphParamsType, NodeType } from '../../types/Graph';
 import { store } from '../../store/ConfigStore';
 
+const URLSearchParams = require('url-search-params');
```
Comment (jotak) [inline src/components/Nav/NavUtils.tsx]: We use URLSearchParams already in several places but use the native object, so I would say, no need to `require` it.
(if old browser support is wanted at some point for `URLSearchParams`, then anyway we'd have to update every place where it's in use)

Comment (jeremyxu2010) [reply src/components/Nav/NavUtils.tsx]: ok, i will delete the `require`.

**Comment by pilhuhn** [comment]
@jotak Aren't the individual items in the overview clickable?
If so, they lead you to other pages. The patch would need to disable all the links embedded in the various pages(?). 
@jeremyxu2010 Kiosk mode is enabled by calling the Kiali-UI with `kiosk=true` url param?

**Comment by jeremyxu2010** [comment]
> @jotak Aren't the individual items in the overview clickable?
> If so, they lead you to other pages. The patch would need to disable all the links embedded in the various pages(?).
> @jeremyxu2010 Kiosk mode is enabled by calling the Kiali-UI with `kiosk=true` url param?

Yeah, my system embed Kiali's page through IFRAME,  only some URL parameters can be passed.

**Comment by rhqci** [comment]
Can one of the admins verify this patch?

**Comment by mtho11** [comment]
@jeremyxu2010 looks like this needs rebasing

**Comment by cfcosta** [comment]
Oh, and of course, it needs to be rebased :)

File: src/pages/Graph/GraphPage.tsx [hunk 2]
```diff
@@ -203,10 +208,16 @@ export default class GraphPage extends React.Component<GraphPageProps, GraphPage
       injectServiceNodes: this.props.injectServiceNodes
     };
 
+    const urlParams = new URLSearchParams(window.location.search);
+    let conStyle = containerStyle;
```
[the remark this answers is not in the document]
Comment (jeremyxu2010) [reply src/pages/Graph/GraphPage.tsx]: ok,  i will improve these code.

**Comment by jeremyxu2010** [comment]
> Oh, and of course, it needs to be rebased :)

ok, i will rebase onto upstream's master

**Comment by pilhuhn** [comment]
Thank you @jeremyxu2010 !

I wonder if you could send a followup PR to README.adoc to document this new feature?

**Comment by abonas** [comment]
was this documented in readme? I can't find it

**Comment by israel-hdez** [comment]
It is not documented, AFAIK.

**Comment by abonas** [comment]
then it shall be @israel-hdez ;)
https://issues.jboss.org/browse/KIALI-3229

**Pull request merged**